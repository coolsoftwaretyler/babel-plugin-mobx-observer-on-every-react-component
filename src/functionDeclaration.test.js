import { expect, test, describe} from "bun:test";
import { transform } from "@babel/core";
import pluginSyntaxJsx from "@babel/plugin-syntax-jsx";
import pluginTransReactJsx from "@babel/plugin-transform-react-jsx";
import { autoObserverPlugin } from "./index.mjs";

const runTransform = (input) => {
  return transform(input, {
    plugins: [pluginSyntaxJsx, pluginTransReactJsx, autoObserverPlugin],
  });
};

describe("Mobx Observer Babel Plugin", () => {
  describe("function declaration", () => {
    describe("when it is not a React component", () => {
      test("it does nothing", () => {
        const source = `function MyFunctionDeclaration() { return 0 }`;
        const out = runTransform(source);

        expect(out.code).toMatchSnapshot();
      });
    });
    describe("when it is a React component", () => {
      describe("when it is not wrapped in observer", () => {
        test("it wraps the component in observer", () => {
          const source = `function MyComponentDeclaration() { return <div>Hello World</div>; }`;
          const out = runTransform(source);

          expect(out.code).toMatchSnapshot();
        });
      });
    });
  });
});
