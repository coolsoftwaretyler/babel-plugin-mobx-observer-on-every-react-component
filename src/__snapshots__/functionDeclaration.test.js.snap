// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobx Observer Babel Plugin function declaration when it is not a React component it does nothing 1`] = `
"function MyFunctionDeclaration() {
  return 0;
}"
`;

exports[`Mobx Observer Babel Plugin function declaration when it is a React component when it is not wrapped in observer it wraps the component in observer 1`] = `
"import { observer } from "mobx-react";
observer(function MyComponentDeclaration() {
  return /*#__PURE__*/React.createElement("div", null, "Hello World");
});"
`;
